ast/TreeEquality.cc

```diff
@@ -7,16 +8,19 @@ namespace sorbet::ast {
 
 namespace {
 
-bool structurallyEqual(const void *avoid, const ExpressionPtr &tree, const ExpressionPtr &other, bool root = false);
+bool structurallyEqual(const core::GlobalState &gs, const void *avoid, const ExpressionPtr &tree,
+                       const ExpressionPtr &other, const std::shared_ptr<spdlog::logger> logger,
```

Comment:
Can we take the logger by const reference instead?

As it is right now, this requires atomic incr/decr operations for every recursive call to `structurallyEqual` to do the reference counting.

Reply:
Even more than that, if you don't want to deal with threading the `logger` through all of these calls, you can use the `fatalLogger` global variable, like we do in `Loc.cc`.